arm-servicebus/2014-09-01/swagger/servicebus.json

```diff
@@ -0,0 +1,974 @@
+        },
+        "region": {
+          "externalDocs": {
+            "url": ""
+          },
+          "type": "string",
+          "description": "Specifies the targeted region in which the namespace should be created. It can be any of the following values: Australia EastAustralia SoutheastCentral USEast USEast US 2West USNorth Central USSouth Central USEast AsiaSoutheast AsiaBrazil SouthJapan EastJapan WestNorth EuropeWest Europe"
+        },
+        "status": {
+          "externalDocs": {
+            "url": ""
+          },
+          "type": "string",
+          "description": "Status of the namespace. It can be any of these values:1 = Created/Active2 = Creating3 = Suspended4 = Deleting"
+        },
+        "createdAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time the namespace was created."
+        },
+        "serviceBusEndpoint": {
+          "externalDocs": {
+            "url": ""
+          },
+          "type": "string",
+          "description": "Endpoint you can use to perform ServiceBus operations."
+        },
+        "subscriptionId": {
+          "type": "string",
+          "description": "The Id of the Azure subscription associated with the namespace."
+        },
+        "scaleUnit": {
+          "type": "string",
+          "description": "ScaleUnit where the namespace gets created"
+        },
+        "enabled": {
+          "type": "boolean",
+          "description": "Whether or not the namespace is currently enabled."
+        },
+        "critical": {
+          "type": "boolean",
+          "description": "Whether or not the namespace is set as Critical."
+        },
+        "namespaceType": {
+          "type": "string",
+          "description": "Gets or sets the namespace type.",
+          "enum": [
+            "Messaging",
+            "NotificationHub",
+            "EventHub"
+          ],
+          "x-ms-enum": {
+            "name": "NamespaceType",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Namespace properties."
+    },
+    "RegenerateKeysParameters": {
+      "properties": {
+        "Policykey": {
+          "type": "string",
+          "description": "Key that needs to be regenerated ."
+        }
+      },
+      "description": "Parameters supplied to the Regenerate Namespace Auth Rule."
+    },
+    "NamespaceCreateOrUpdateParameters": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "Gets or sets Namespace data center location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets Namespace tags."
+        },
+        "properties": {
+          "$ref": "#/definitions/NamespaceProperties",
+          "description": "Gets or sets properties of the Namespace."
```

Comment:
Apply the extension `"x-ms-client-flatten" : true`. This will bring all the NamespaceProperties one level up, thus making it easier for user to construct the parameters. Take a look at the documentation https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-client-flatten for more info